### PR TITLE
[iOS] Fix NullReferenceException when MediaElement methods are called before setting Source

### LINKED ISSUE #414: [Bug] MediaElement throws NullReferenceException paper cut
### Description

`MediaElement` throws a `NullReferenceException` when calling the `Play`, `Pause`, and `Stop` methods when a `Source` hasn't been specified. Obviously this can be coded around by only calling those methods dependent upon the `CurrentState` property. So this is a paper cut rather than a bug.

Perhaps the UWP approach should be adopted, which is that "Any calls to Play, Pause, and Stop methods that occur before the MediaOpened event is raised are ignored."

### PR DESCRIPTION
**Pull request by daniel-shuy**
### Description of Change ###

This PR updates the iOS `MediaElementRenderer` to ignore `Play()`/`Pause()`/`Stop()` events if [AVPlayerItem](https://developer.apple.com/documentation/avfoundation/avplayeritem) is not yet set (using [constructor](https://developer.apple.com/documentation/avfoundation/avplayer/1387104-init) or [AVPlayer.replaceCurrentItem(AVPlayerItem?)](https://developer.apple.com/documentation/avfoundation/avplayer/1390806-replacecurrentitem), which is called when the Source is updated).

### Bugs Fixed ###

- Fixes #414

### API Changes ###

None

<!-- List all API changes here (or just put None), example:

Added: 
 
- `string Class.Property { get; set; }`
- `void Class.Method();`

Changed:

 - `object Cell.OldPropertyName` => `object Cell.NewPropertyName`
 
If there is an entirely new API, then you can use a more verbose style:

```csharp
public static class NewClass {
    public static int SomeProperty { get; set; }
    public static void SomeMethod(string value);
}
``` -->

### Behavioral Changes ###

<!-- Describe any non-bug related behavioral changes that may change how users app behaves when upgrading to this version of the codebase. -->

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=470) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=470) before we can accept your contribution.<br/><br/>:x: daniel-shuy [sign now](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=470)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/xamarin/XamarinCommunityToolkit?pullRequest=470) it.</sub>

File: XamarinCommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs [hunk 1]
```diff
@@ -307,6 +307,12 @@ protected virtual void Play()
 
 		void MediaElementStateRequested(object sender, StateRequested e)
 		{
+			// if no Source is set, do nothing
+			if (avPlayerViewController.Player?.CurrentItem == null)
+            {
+				return;
+            }
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs:314]:
```suggestion
			if (avPlayerViewController.Player?.CurrentItem == null)
				return;
```

**Comment by jfversluis** [comment]
@daniel-shuy thank you for this! Have you actually tried to reproduce this? 
I tried to reproduce to verify this fix, but I can't seem to be able to, so maybe it was already fixed somehow? Or am I overlooking something?

I took the current main branch, removed the `Source` from the `MediaElement` and added a button that calls the `Play` method on the `MediaElement` but nothing happens with the current code as well.

**Comment by daniel-shuy** [comment]
Sorry, I made the fix based on what was reported in #414 without testing it myself, but looking closer at it I realize that the bug is actually in `Xamarin.Forms`' `MediaElement`, and it seems like the `XamarinCommunityToolkit`'s `MediaElement` doesn't have this same issue.

**Pull request closed without merging**